### DISCUSSION TITLE
How to stop all services in ISO-2.3.10

### DISCUSSION THREAD

**Discussion by ldceven** [Security-Onion-Solutions/securityonion#2058] [Unsupported Versions] [answered]
I am using the so-2.3.10 version new , and I want to add a hard disk to /nsm. How do I stop all services in the so-2.3.10 version? The sudo so-stop command cannot be executed.
thanks.

"Please provide an argument by running like so-stop , or by using the component-specific script.
Ex. so-stop filebeat, or so-filebeat-stop
"

**Comment by weslambert** [comment, marked as answer]
Right now, we don't have a wrapper script to stop them all at once, so you would need to loop over them, like;

`for i in $(docker container ls -q  --filter name=so-* ); docker stop $i &&
docker rm $i; done`.

We can address this as part of https://github.com/Security-Onion-Solutions/securityonion/issues/383

**Reply by ldceven** [reply]
After docker stop&&docker rm, how do I restart the docker container?

**Reply by stijnpieters** [reply]
`so-start all`
should work
But at the end of the day that runs
`salt-call state.highstate`
which is also a command you could run

**Reply by DashCarlos** [reply]
I used your script but they immediately start to get up again, any idea? 
![imagen_2020-12-03_104435](https://user-images.githubusercontent.com/31094207/100992503-892ec200-3554-11eb-985e-366582b9d229.png)

**Reply by stijnpieters** [reply]
It could be highstate was applied directly after you deleted all the containers. 
Highstate is applied every 15 minutes.

**Reply by DashCarlos** [reply]
> It could be highstate was applied directly after you deleted all the containers.
> Highstate is applied every 15 minutes.

Is there a way to stop the "highstate"?

**Reply by stijnpieters** [reply]
DISCLAIMER:I'm not super experienced with Saltstack, all I know of it has been because Security Onion uses it. I have not tested this myself, but it looks pretty okay lol.

First you'll need the salt key of your node: `salt-key`, this lists all nodes. Look for the one you want to disable and copy the key. (I assume this will just be 1 key in your case)

I think `sudo salt '<salt key>' schedule.disable` should disable the schedule until enabled again.
To re-enable the schedule, run `sudo salt '<salt key>' schedule.enable`
